packages/driver/src/cy/commands/actions/type.coffee

```diff
@@ -112,8 +112,10 @@ module.exports = (Commands, Cypress, cy, state, config) ->
       if chars is ""
         $utils.throwErrByPath("type.empty_string", { onFail: options._log })
 
-      if options.$el.attr("maxlength") and chars.length > options.$el.attr("maxlength")
-        chars = chars.slice(0, options.$el.attr("maxlength"))
+      if options.$el.attr("maxlength")
+        maxLength = parseInt(options.$el.attr("maxlength"))
```

Comment:
This should use the radix param for parseInt: `parseInt(options.$el.attr("maxlength"), 10)`

Reply:
Oh right thanks, I often forget that it should be included.